packages/@glimmer/runtime/test/initial-render-test.ts

```diff
@@ -503,10 +504,18 @@ class RehydratingComponents extends AbstractRehydrationTests {
   }
 
   assertServerComponent(html: string, attrs: Object = {}) {
+    let element: HTMLElement;
+
     if (this.testType === 'Dynamic') {
-      assertEmberishElement(this.element.childNodes[3] as HTMLElement, 'div', attrs, html);
+      element = this.element.childNodes[3] as HTMLElement;
```

Comment:
This is not a blocker, but this seems a little brittle. Do you have any ideas for how we could find the right element without special casing offsets?

Reply:
Ah, that sounds like a likely explanation. Again, I know you're inheriting this bit of code, so please don't consider it urgent or a blocker, unless there's an obvious, trivial improvement we can sneak in.

Reply:
OK, so I confirmed our suspicion here. When `{{component` is rendered in SSR we add comment node markers for it and therefore have to account for that when reading which of the childNodes we care about. I cleaned up the logic a little and left a comment to that effect.